### PR TITLE
fix(helper): update _interopRequireWildcard

### LINKED ISSUE #4284: Dynamically importing JSON module containing simple string in CJS loads wrong shape
### Describe the bug

When a JSON module is imported using `import()` and it contains a primitive, SWC's output is not spec-compliant, because when traversing keys, it doesn't check whether the object is a primitive.

### Input code

```js
import("./data.json").then(console.log);
```

`data.json`:

```json
"foo"
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript"
    },
    "target": "es2017"
  },
  "module": {
    "type": "commonjs"
  }
}
```

[Playground](https://play.swc.rs/?version=1.2.164&code=H4sIAAAAAAAAA8vMLcgvKtFQ0tNPSSxJ1Msqzs9T0tQryUjN00jOzyvOz0nVy8lP17TmAgAoHhbMKQAAAA%3D%3D&config=H4sIAAAAAAAAA0WMSwrDMAxE76K1t93kDj2EcJXg4B8aBWqM714bUrKSmPdmOp3wtHWqrBBdH1o2%2FtJG4hPDa6hGbmoz2jlChiNjPcSWgtdksRTITR2lkMPe1pIvqaoAD%2BJ8xL855lAqn2sFnazVeVcllXyCxjN0lwPet216yfgBCKaI9boAAAA%3D)

### Expected behavior

It should log `{ default: 'foo' }`.

### Actual behavior

It logs `{ '0': 'f', '1': 'o', '2': 'o', default: 'foo' }`.

Here's [Babel's playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=JYWwDg9gTgLgFAIgHQHoAmBDGGkCsDOEAdggJRIwAWApkXAMbGEA21SzEA5qQNwBQQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-3%2Ctypescript&prettier=true&targets=&version=7.17.9&externalPlugins=%40babel%2Fplugin-transform-spread%407.16.7&assumptions=%7B%7D). You can see they have an additional

```js
if (obj === null || (typeof obj !== "object" && typeof obj !== "function")) {
  return { default: obj };
}
```

### Version

1.2.164

### Additional context

Originally reported in https://github.com/swc-project/swc-loader/issues/43

### PR DESCRIPTION
**Pull request by RiESAEX**
**Description:**
since we don't have `importInterop: "node"`, I didn't update to the latest version in babel.

**BREAKING CHANGE:**

**Related issue (if exists):**
fix #4284

### CODE REVIEW

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you update https://github.com/swc-project/swc/blob/48b427e403552f56cb4c26dfcd55d7c632a77781/packages/swc-helpers/src/_interop_require_wildcard.js too?

**Comment by RiESAEX** [comment]
I guess you mean the file in swc_ecma_transforms_base?

**Comment by kdy1** [comment]
Ah, yes sorry

File: crates/swc_ecma_transforms_base/src/helpers/_interop_require_wildcard.js [hunk 1]
```diff
@@ -1,24 +1,42 @@
-function _interopRequireWildcard(obj) {
+function _getRequireWildcardCache() {
+  if (typeof WeakMap !== "function") return null;
+  var cache = new WeakMap();
+  _getRequireWildcardCache = function () { return cache; };
+  return cache;
+}
+export default function _interopRequireWildcard(obj) {
```
[the remark this answers is not in the document]
Comment (RiESAEX) [reply crates/swc_ecma_transforms_base/src/helpers/_interop_require_wildcard.js]: sorry, I just copy and paste it directly. removed.

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_transforms_base

**Comment by kdy1** [comment]
I'm not sure why we inline helper in the test, but can you update the test?

**Review by kdy1 — APPROVED** [review]

**Pull request merged**